.circleci/config.yml

```diff
@@ -9,17 +9,23 @@ jobs: # a collection of steps
       - checkout # special step to check out source code to working directory
 
       - restore_cache: # special step to restore the dependency cache
-          key: dependency-cache-{{ checksum "package.json" }}-yarn2
+          key: dependency-cache-{{ checksum "yarn.lock" }}
```

Comment:
This is dangerous as not always do we check in a modified "yarn.lock" file and in general I think it is ok not to keep it always checked in. It tends to get merge conflicts and stuff. Why did you change it?

Reply:
Two reasons:
1. checksum only the root package.json and not all the nested package.json is incorrect
2. We care only about installed packages when caching node_modules, we don't really care about other package.json changes that don't affect the installed packages ( added scripts etc.)

I took example from circleci https://circleci.com/docs/2.0/sample-config/ using yarn.lock